### PR TITLE
Improve dynamic validation

### PR DESCRIPTION
**Pull request by martinmladenov**
# Description of changes made
Added support for cached test cases and improved UX

### CODE REVIEW

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt [hunk 1]
```diff
@@ -111,6 +111,7 @@ class TestCaseDisplayService(private val project: Project) {
         selectAllButton.addActionListener { toggleAllCheckboxes(true) }
         deselectAllButton.addActionListener { toggleAllCheckboxes(false) }
         toggleJacocoButton.addActionListener { toggleJacocoCoverage() }
+        removeAllButton.addActionListener { removeAllTestCases() }
```
Comment (jzelenjak) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt:114]: Didn't i add that before?

**Review by jzelenjak — COMMENTED** [review]
Validation works for me: passing tests, failing tests, compilation errors. Both on fresh tests, as well as on cached tests.
However, i don't understand anything in the code, so i think it is better if someone who can be sure that the code makes sense will approve.
Nevertheless, the code seems to be working on my machine.

**Review by kirilvasilev16 — APPROVED** [review]
Nice fix. It works on Windows!

**Review by bolkhod — APPROVED** [review]
From my gatherings, everything seems to work. Good job.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt (hunk 1 above)
Comment (martinmladenov) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseDisplayService.kt:114]: Yes, but someone broke it during a merge.

**Pull request merged**